src/sdam/state/mod.rs

```diff
@@ -214,21 +215,26 @@ impl Topology {
         // doesn't check the fields of an Unknown server, and we only return Unknown server
         // descriptions when errors occur. Once we implement SDAM monitoring, we can
         // properly inform users of errors that occur here.
-        let _ = state_clone.update(server_description, self.clone());
+        let _ = state_clone.update(server_description, self.common.options.clone());
 
-        if old_description == state_clone.description {
-            return false;
-        }
+        match old_description.diff(&state_clone.description) {
```

Comment:
because `add_new_server` isn't starting the monitors anymore, we need to know which servers were added by the latest update to the topology. This diffing might be a little over engineered, but I figured it might also be useful for SDAM monitoring in the future.